pkg/handlers/generic/lifecycle/config/cm.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2023 D2iQ, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package config
+
+import (
+	"context"
+	"fmt"
+
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type HelmConfig struct {
+	cl          ctrlclient.Reader
+	cmName      string
+	cmNamespace string
+}
+
+type HelmSettings struct {
+	HelmChartName       string `yaml:"defaultChartName"`
+	HelmChartVersion    string `yaml:"defaultChartVersion"`
+	HelmChartRepository string `yaml:"defaultRepositoryUrl"`
+}
```

Comment:
I'm a little confused about these struct names, "settings" and "configs" are pretty similar

WDYT about something like:

```
type FromConfigMap struct {
	cl          ctrlclient.Reader
        Name      string
	Namespace string
}

func NewFromConfigMapGetter(cmName, cmNamespace string, cl ctrlclient.Reader) *FromConfigMap

type HelmChart struct {
    	Name       string `yaml:"defaultChartName"`
	Version    string `yaml:"defaultChartVersion"`
	Repository string `yaml:"defaultRepositoryUrl"`
}

func (h *FromConfigMap) For(
	ctx context.Context,
	addon string,
) (*HelmChart, error) {
```

Reply:
HelmChart is for sure better than helmSettings.